src/Symfony/Component/Security/Core/Tests/Authentication/Token/SwitchUserTokenTest.php

```diff
@@ -72,5 +72,53 @@ public function getSalt()
         $token = new SwitchUserToken($impersonated, 'bar', 'provider-key', ['ROLE_USER', 'ROLE_PREVIOUS_ADMIN'], $originalToken);
         $token->setUser($impersonated);
         $this->assertTrue($token->isAuthenticated());
+
+        $impersonator = new class() implements UserInterface {
```

Comment:
I think it's better to create a separate unit for this

Reply:
Now is it problem in User equals test: User without role now is equivalent with user with role "ROLE", but opposite is not true.
https://github.com/gorshkov-ag/symfony/blob/9de09f0cfbfed2af1f7ae0a63597171615a51ac3/src/Symfony/Component/Security/Core/Tests/User/UserTest.php#L110-L120
120 line case.